✨ feat: Add Spark model provider

hezhijie0327 opened the pull request:
#### 💻 变更类型 | Change Type

<!-- For change type, change [ ] to [x]. -->

- [X] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] ⚡️ perf
- [ ] 📝 docs

#### 🔀 变更说明 | Description of Change

基于 #2804 实现支持讯飞星火大模型

~~讯飞的 API 比较特殊，`APIKey:APISecret`~~ 8月改用 `APIPassword`

https://www.xfyun.cn/doc/spark/HTTP%E8%B0%83%E7%94%A8%E6%96%87%E6%A1%A3.html#_3-%E8%AF%B7%E6%B1%82%E8%AF%B4%E6%98%8E

<!-- Thank you for your Pull Request. Please provide a description above. -->

#### 📝 补充信息 | Additional Information
![image](https://github.com/lobehub/lobe-chat/assets/58100052/a0e68e5e-cd3c-4366-87dd-5f1c530f7d4e)
![image](https://github.com/lobehub/lobe-chat/assets/58100052/29d8376b-2cde-4bf9-a68e-5dfe703542ac)
![image](https://github.com/lobehub/lobe-chat/assets/58100052/b7bfca66-a499-4053-a7aa-8fc9e34c0929)
![image](https://github.com/user-attachments/assets/0789b7b1-0809-40a2-9a85-418adadb3edd)

<!-- Add any other context about the Pull Request here. -->

hezhijie0327 committed:
✨ feat: Add Spark model provider

arvinxx commented:
配置页面是不是应该改成 APIKey 和 APISecret 分开配置比较合适？不然用户很有可能会配错

hezhijie0327 committed:
🔨 chore: split Spark API Key & Spark Secret

hezhijie0327 commented:
逻辑都通了，自动将 APIKey 与 API Secret 合并成完整 API  

但是遇到个问题，这个文言显示搞不明白
![image](https://github.com/lobehub/lobe-chat/assets/58100052/10436021-7188-478b-87ec-0cdfb18a5322)

arvinxx commented:
是 locale 吗？ 参考这里补充 i18n:

https://github.com/lobehub/lobe-chat/blob/main/src/locales/default/modelProvider.ts#L22-L48

arvinxx commented on src/app/(main)/settings/llm/ProviderList/Spark/index.tsx:
```diff
@@ -0,0 +1,58 @@
+  const theme = useTheme();
+  return (
+    <Spark.Combine
+      color={theme.isDarkMode ? theme.colorText : Spark.colorPrimary}
```
这里感觉不需要加这个color，去掉看下效果？加上color以后变成单色的了

hezhijie0327 replied:
改好了
![image](https://github.com/lobehub/lobe-chat/assets/58100052/6c3c4ebd-d55e-4ac5-a202-76994050d1f7)

hezhijie0327 committed:
💄 style: update Spark icon

hezhijie0327 committed:
💄 style: update Spark icon size

hezhijie0327 commented:
> 是 locale 吗？ 参考这里补充 i18n:
> 
> https://github.com/lobehub/lobe-chat/blob/main/src/locales/default/modelProvider.ts#L22-L48

放进去了 但是貌似没调用 不清楚哪里写错了

arvinxx commented:
@hezhijie0327  dev没用还是prod没用？prod没用的话需要 npm run i18n 去自动生成其他语言的 locale

hezhijie0327 commented:
> @hezhijie0327 dev没用还是prod没用？prod没用的话需要 `npm run i18n` 去自动生成其他语言的 locale

dev 下本地 Build，这应该是缺省值吧，缺少 locale 的话应该也能显示的吧。我先跑下 `npm run i18n` 看看

arvinxx commented:
> dev 下本地 Build，这应该是缺省值吧，缺少 locale 的话应该也能显示的吧

没，故意做成不显示的，不然发布版本默认兜底成中文的话，没法让开发者意识到要跑 locale： https://github.com/lobehub/lobe-chat/blob/main/src/locales/create.ts#L21-L25

这样一来 build 过的生产版本，如果没有 locale 必然就只会显示 locale key ，更容易发现没国际化的问题。

hezhijie0327 commented:
> > dev 下本地 Build，这应该是缺省值吧，缺少 locale 的话应该也能显示的吧
> 
> 没，故意做成不显示的，不然发布版本默认兜底成中文的话，没法让开发者意识到要跑 locale： https://github.com/lobehub/lobe-chat/blob/main/src/locales/create.ts#L21-L25
> 
> 这样一来 build 过的生产版本，如果没有 locale 必然就只会显示 locale key ，更容易发现没国际化的问题。

懂了，其实一直都没 BUG 哈哈哈
![image](https://github.com/lobehub/lobe-chat/assets/58100052/ac2858d6-1958-4f0a-80e6-47ade4ae4be5)

hezhijie0327 committed:
💄 style: update Spark icon in ProviderAvatar

hezhijie0327 commented:
FYI. 已修复 `provider Avatar` 问题

hezhijie0327 committed:
🔨 chore: update Spark models

hezhijie0327 committed:
🔨 chore: update Spark models

hezhijie0327 commented:
暂时禁用 Lite 和 V2.0 模型，开了工单给讯飞，其余模型均可正常工作
![image](https://github.com/lobehub/lobe-chat/assets/58100052/905a659d-1bf7-4d5e-82d8-19322a12ddc1)
感觉他们 API 没做好，模型处任意填写都能过然后 Fallback 到 Pro 模型
```
curl -i -k -X POST 'https://spark-api-open.xf-yun.com/v1/chat/completions' \
--header 'Authorization: Bearer apiKey:apiSecret' \
--header 'Content-Type: application/json' \
--data '{
    "model":"general", // general, generalv2
    "messages": [
        {
            "role": "user",
            "content": "Hi"
        }
    ]
}'
```

hezhijie0327 committed:
Merge branch 'dev' into spark

arvinxx commented:
Fallback这个行为和 google 有的一拼…

hezhijie0327 committed:
💄 style: fixed Spark 4.0 Ultra model icon display

hezhijie0327 committed:
🔨 chore: update Spark models info

hezhijie0327 commented:
移除 `functionCall` 支持，模型内建插件并不支持外部插件引入

hezhijie0327 committed:
🔨 chore: update Spark models tokens info

hezhijie0327 commented:
1. `tokens`参数即最大上下文长度官网没数据，找讯飞技术确认了下为 `8K` [已更新]
![image](https://github.com/lobehub/lobe-chat/assets/58100052/a8559119-4c7c-42e9-80e4-3efc1251dbb4)
2. `general` 和 `generalv2` 他们还在排查中（但是说实话他们的技术支持回复稍微有点一言难尽）

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 commented:
`general`和 `generalv2` 这两个模型确认了，应下发无限 Token 的 Lite 没有下发到 API 中引起的
![image](https://github.com/lobehub/lobe-chat/assets/58100052/d7230444-94b7-456b-951f-8bf9426fc05a)

`generalv2`淘汰了，只保留四个产品线 lite, pro, max, ultra 即 general, generalv3, generalv3.5 和 4.0Ultra

hezhijie0327 committed:
🔨 chore: update Spark models info

hezhijie0327 commented:
讯飞把 Lite Token 数量问题修好了
![image](https://github.com/lobehub/lobe-chat/assets/58100052/080f146e-d783-49f1-b016-00c1f78d41f6)

但是执行时报了以下错误，这个是什么引起的？
```
[stream start] 2024-7-3 10:43:25.692
[chunk 0] 2024-7-3 10:43:25.704
{"code":10004,"message":"SchemaCheckError:(18:43:25.205) '$.header.uid' length must be less or equal than 32; ","sid":"cha000b1ecf@dx19078312593b8f2532","id":"cha000b1ecf@dx19078312593b8f2532","created":1720003405}
```

hezhijie0327 commented:
我试了一个 Demo，没啥问题，但是跑到 LobeChat 就报错，很神奇
```
from openai import OpenAI
client = OpenAI(
        api_key="apiKey:apiSecret",
        base_url = 'https://spark-api-open.xf-yun.com/v1' # 指向讯飞星火的请求地址
    )
completion = client.chat.completions.create(
    model='general',
    messages=[
        {
            "role": "user",
            "content": '你是谁'
        }
    ],
    stream=True
)
for chunk in completion:
    print(chunk)
    print(chunk.choices[0].delta.content)
```
```
ChatCompletionChunk(id='cha000b4308@dx190788783bfb8f2532', choices=[Choice(delta=ChoiceDelta(content='\n', function_call=None, role='assistant', tool_calls=None), finish_reason=None, index=0, logprobs=None)], created=1720009065, model=None, object=None, service_tier=None, system_fingerprint=None, usage=None, code=0, message='Success', sid='cha000b4308@dx190788783bfb8f2532')


ChatCompletionChunk(id='cha000b4308@dx190788783bfb8f2532', choices=[Choice(delta=ChoiceDelta(content='\n我是一个', function_call=None, role='assistant', tool_calls=None), finish_reason=None, index=0, logprobs=None)], created=1720009065, model=None, object=None, service_tier=None, system_fingerprint=None, usage=None, code=0, message='Success', sid='cha000b4308@dx190788783bfb8f2532')

我是一个
ChatCompletionChunk(id='cha000b4308@dx190788783bfb8f2532', choices=[Choice(delta=ChoiceDelta(content='AI语言模型，', function_call=None, role='assistant', tool_calls=None), finish_reason=None, index=0, logprobs=None)], created=1720009065, model=None, object=None, service_tier=None, system_fingerprint=None, usage=None, code=0, message='Success', sid='cha000b4308@dx190788783bfb8f2532')
AI语言模型，
ChatCompletionChunk(id='cha000b4308@dx190788783bfb8f2532', choices=[Choice(delta=ChoiceDelta(content='可以回答您的问题和', function_call=None, role='assistant', tool_calls=None), finish_reason=None, index=0, logprobs=None)], created=1720009066, model=None, object=None, service_tier=None, system_fingerprint=None, usage=None, code=0, message='Success', sid='cha000b4308@dx190788783bfb8f2532')
可以回答您的问题和
ChatCompletionChunk(id='cha000b4308@dx190788783bfb8f2532', choices=[Choice(delta=ChoiceDelta(content='提供帮助。', function_call=None, role='assistant', tool_calls=None), finish_reason=None, index=0, logprobs=None)], created=1720009066, model=None, object=None, service_tier=None, system_fingerprint=None, usage=CompletionUsage(completion_tokens=19, prompt_tokens=10, total_tokens=29), code=0, message='Success', sid='cha000b4308@dx190788783bfb8f2532')
提供帮助。
```

hezhijie0327 commented:
找到 Root Cause 了，Spark API 设计问题，只有 Lite 模型加了 User 长度限制...

https://github.com/lobehub/lobe-chat/blob/7fa3b425464b75d0f3aed09e9dd162341f7b554e/src/libs/agent-runtime/utils/openaiCompatibleFactory/index.ts#L88 
```
curl -i -k -X POST 'https://spark-api-open.xf-yun.com/v1/chat/completions' \
--header 'Authorization: Bearer apiKey:apiSecret' \
--header 'Content-Type: application/json' \
--data '{
    "model":"general",
    "stream": true,
    "frequency_penalty": 0,
    "presence_penalty": 0,
    "temperature": 0.6,
    "top_p": 1,
    "messages": [
        {
            "content": "Hi",
            "role": "user"
        }
    ],
    "user": "b03ff615-8e78-49cf-85a1-8af9081be721" // 长度为 36 个字符，Lite 模型限制在 32 个字符内
}'
```

hezhijie0327 committed:
🐛 fix: fixed "'$.header.uid' length must be less or equal than 32" with Spark Lite

hezhijie0327 commented:
`Lite` 模型的问题搞定了
![image](https://github.com/lobehub/lobe-chat/assets/58100052/019445ca-d8ea-4891-8759-713de48fd16a)

和科大工程师确认了下 Spark 全系暂不支持 `user` 参数，其余模型可用也只是预留参数，并无实际用途
![image](https://github.com/lobehub/lobe-chat/assets/58100052/2fb7b861-d8f2-4216-b932-529fd2a42ebf)
 
Solution: 引入了一个 `noUserId` 的 boolean 变量，用于控制是否在 POST 中添加 `user`，默认保持现有逻辑进行添加，当为 `true` 时，忽略 `user`

arvinxx commented on src/libs/agent-runtime/utils/openaiCompatibleFactory/index.ts:
```diff
@@ -85,7 +86,10 @@ export const LobeOpenAICompatibleFactory = ({
-          { ...postPayload, user: options?.user },
+          {
+            ...postPayload,
+            ...(chatCompletion?.noUserId ? {} : { user: options?.user }) 
```
这一行得补一个单测

hezhijie0327 replied:
加上了，测试项目如下：
```
1. noUserId: true -> not contain "testUser"
2. noUserId: false -> contain "testUser"
3. 不指定 noUserId -> contain "testUser"
```

arvinxx commented:
<img width="320" alt="image" src="https://github.com/lobehub/lobe-chat/assets/28616219/0577f117-5a91-49c2-8abe-f6d5b6505614">

这个 general 的tag没有 icon 么？

hezhijie0327 commented:
> <img alt="image" width="320" src="https://private-user-images.githubusercontent.com/28616219/346308777-0577f117-5a91-49c2-8abe-f6d5b6505614.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAzNDMwMjYsIm5iZiI6MTcyMDM0MjcyNiwicGF0aCI6Ii8yODYxNjIxOS8zNDYzMDg3NzctMDU3N2YxMTctNWE5MS00OWMyLThhYmUtZjZkNWI2NTA1NjE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA3VDA4NTg0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmNTE0ZmE1NGI4ZGU4MjAzNjQ5Nzc5NDAxYjYwOTkyMGJlZWI1MzA5YTk0YTE0YWNkYmEwY2IxMjdmNGZmMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wS6effPhhqYloBn2bpTaAdE9xU8aHNslWAId2FhGmTE">
> 这个 general 的tag没有 icon 么？

是指这个 icon 嘛？我看看 Zhipu 的是哪里设置的，template 里没要求设置 
![image](https://github.com/lobehub/lobe-chat/assets/58100052/0b6bc24d-cbb8-4876-97a2-157ac28fcb9b)

arvinxx commented:
@hezhijie0327 可能是我 template 里漏了…

hezhijie0327 commented:
`src/components/ModelTag/ModelIcon.tsx` 这里对吧

hezhijie0327 committed:
💄 style: fix model tag icon missing

hezhijie0327 committed:
🐛 fix: fix typo in ModelIcon

hezhijie0327 commented:
Tag Icon 问题修掉了
![image](https://github.com/lobehub/lobe-chat/assets/58100052/fda67211-8847-485c-a298-b09a7d4432ae)
对 Baichuan 模型 Tag 的问题也修了，原先 model 没有进行小写转换的预处理
![image](https://github.com/lobehub/lobe-chat/assets/58100052/a996a11f-952a-4b81-99f7-eec183c31b4f)

hezhijie0327 committed:
🔨 chore: add unit test for noUserId

arvinxx commented:
@hezhijie0327 为啥我测下来一直显示 loading 态呢，状态码已经 200 了

<img width="1904" alt="image" src="https://github.com/lobehub/lobe-chat/assets/28616219/20c4e96a-9be0-41b6-bfed-e5ab4e23ebd6">

hezhijie0327 commented:
我试了下都正常呀
![image](https://github.com/lobehub/lobe-chat/assets/58100052/e1d74f23-5fe5-4e22-bb5a-7746c6f33a43)
看下 Console 里有使用记录不 `https://console.xfyun.cn/services/bm4`
![image](https://github.com/lobehub/lobe-chat/assets/58100052/a5cb7edf-133e-4ba0-9bda-fa4d64fe53de)

hezhijie0327 commented:
这里每个都领一下看看 `https://xinghuo.xfyun.cn/sparkapi`
![image](https://github.com/lobehub/lobe-chat/assets/58100052/70606bc8-6a2d-4ac5-a507-a1f1dd04ce90)

hezhijie0327 commented:
我的感觉是 Token 没领取，他的每个模型和 AppID 是单独计费的，
1. general3 能用是 pro 模型暂时免费
2. Lite 的你要看下是否为无限，他们有一批注册的账户没有自动分配显示为0，这个只能开工单修复，这个之前测试的时候吃过亏
![image](https://github.com/lobehub/lobe-chat/assets/58100052/f775e530-5a33-4d82-a304-04eccf00b441)
![image](https://github.com/lobehub/lobe-chat/assets/58100052/3f3c0970-4420-46c4-8f75-2f2b722e21d2)

hezhijie0327 commented:
@arvinxx 应该是 Token 的问题了，我在同一个测试页面也能过 
![image](https://github.com/lobehub/lobe-chat/assets/58100052/59d536ec-266b-4381-bb46-7d68dc6c01d9)

arvinxx commented:
@hezhijie0327 如果是token的问题，是不是应该提示抛错，而不是 200 正常过？

hezhijie0327 commented:
@arvinxx  Console 显示是 `The user aborted a request.`
![image](https://github.com/lobehub/lobe-chat/assets/58100052/66f62258-3a9c-4280-8596-9a1d81547723)
启用 debug 后发现，在 Stream 流里有error code, 尝试用 handleError 貌似没啥用  
```
[stream start] 2024-7-9 1:9:45.37
[chunk 0] 2024-7-9 1:9:45.46
{"code":11200,"message":"AppIdNoAuthError:(09:09:46.523) tokens.total","sid":"cha000bcc07@dx190950a1d97b8f2532","id":"cha000bcc07@dx190950a1d97b8f2532","created":1720487386}

 ⨯ Error: failed to pipe response
    at pipeToNodeResponse (/opt/lobechat/node_modules/.pnpm/next@14.2.4_@babel+core@7.23.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/pipe-readable.js:126:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async NextNodeServer.runEdgeFunction (/opt/lobechat/node_modules/.pnpm/next@14.2.4_@babel+core@7.23.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/next-server.js:1250:13)
    at async NextNodeServer.handleCatchallRenderRequest (/opt/lobechat/node_modules/.pnpm/next@14.2.4_@babel+core@7.23.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/next-server.js:248:37)
    at async NextNodeServer.handleRequestImpl (/opt/lobechat/node_modules/.pnpm/next@14.2.4_@babel+core@7.23.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/base-server.js:816:17)
    at async invokeRender (/opt/lobechat/node_modules/.pnpm/next@14.2.4_@babel+core@7.23.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/lib/router-server.js:174:21)
    at async handleRequest (/opt/lobechat/node_modules/.pnpm/next@14.2.4_@babel+core@7.23.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/lib/router-server.js:353:24)
    at async requestHandlerImpl (/opt/lobechat/node_modules/.pnpm/next@14.2.4_@babel+core@7.23.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/lib/router-server.js:377:13)
    at async Server.requestListener (/opt/lobechat/node_modules/.pnpm/next@14.2.4_@babel+core@7.23.6_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/server/lib/start-server.js:141:13) {
  [cause]: TypeError: Cannot read properties of undefined (reading '0')
      at $ (/opt/lobechat/.next/server/edge-chunks/603.js:7:5076)
      at Object.transform (/opt/lobechat/.next/server/edge-chunks/603.js:3:2155)
      at invokePromiseCallback (node:internal/webstreams/util:180:10)
      at Object.<anonymous> (node:internal/webstreams/util:185:23)
      at transformStreamDefaultControllerPerformTransform (node:internal/webstreams/transformstream:526:37)
      at transformStreamDefaultSinkWriteAlgorithm (node:internal/webstreams/transformstream:572:10)
      at node:internal/webstreams/transformstream:377:16
      at writableStreamDefaultControllerProcessWrite (node:internal/webstreams/writablestream:1127:5)
      at writableStreamDefaultControllerAdvanceQueueIfNeeded (node:internal/webstreams/writablestream:1242:5)
      at writableStreamDefaultControllerWrite (node:internal/webstreams/writablestream:1116:3)
}
[stream finished] total chunks: 1

Error: aborted
    at abortIncoming (node:_http_server:806:17)
    at socketOnClose (node:_http_server:800:3)
    at Socket.emit (node:events:532:35)
    at TCP.<anonymous> (node:net:339:12)
    at TCP.callbackTrampoline (node:internal/async_hooks:130:17) {
  code: 'ECONNRESET'
}
```

hezhijie0327 commented:
哎 他们 API 返回的状态值很迷...

1. Stream: true; 错误但是返回 200 success
![image](https://github.com/lobehub/lobe-chat/assets/58100052/004b9ca0-7ea6-4d2d-93c9-86397ad45b88)
2. Stream: false 
![image](https://github.com/lobehub/lobe-chat/assets/58100052/d3a2ca25-75fd-4db4-b951-6f7312c75499)

一个最简单的解法是关闭 stream，然后就能根据现有的 handleError 抓到 500 错误了，本地试了已经 Ok了，我删掉些测试代码 commit 下   
![image](https://github.com/lobehub/lobe-chat/assets/58100052/f059cddb-fccb-48b3-9432-1324366aeeea)

hezhijie0327 committed:
🔨 chore: disable stream mode

hezhijie0327 commented:
@arvinxx 试下新的 Commit

arvinxx commented:
你先改回来吧。这个之前做 minimax 的 provider也有遇到这个问题。之前也不好解决，但我现在知道应该怎么解了。我回头在你这个pr基础上加一下这块逻辑。然后后面可以把minimax 的实现再重构一下

hezhijie0327 committed:
Revert "🔨 chore: disable stream mode" (#25)

This reverts commit 302e01d181a8f949053e04df85d23f69d2149039.

hezhijie0327 commented:
Rollback 了

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
💄 style: add Spark Pro-128K new model

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
✨ feat: Add Spark ENV

hezhijie0327 committed:
🐛 fix: fixed Pro-128k model id, wrong id from official document

![image](https://github.com/user-attachments/assets/7fc3fc73-b460-448c-ad78-4a56d3cae34e)

hezhijie0327 commented:
添加最新发布的`Pro-128k`模型

PS: 讯飞官网的文档还是错的... 模型 ID 应为首字母大写，已开工单反馈
![image](https://github.com/user-attachments/assets/1123f77b-62ff-4263-affe-365bbcd3bb1e)

hezhijie0327 committed:
Merge branch 'main' into spark

arvinxx commented:
https://github.com/lobehub/lobe-chat/discussions/3361

hezhijie0327 committed:
💄style: improve APIKeyForm for Spark

hezhijie0327 committed:
💄 style: improve custom Spark API missing form

hezhijie0327 commented:
拆分 PR:
1. `noUserId` 用于修复 Mistral 模型调用的问题
2. Stepfun 及 Baichuan 模型 TagIcon 丢失的问题

hezhijie0327 committed:
Merge branch 'main' into spark

arvinxx committed:
Merge branch 'main' into spark

arvinxx commented on src/libs/agent-runtime/AgentRuntime.ts:
```diff
@@ -247,6 +249,11 @@ class AgentRuntime {
       }
+
+      case ModelProvider.Spark: {
+        runtimeModel = new LobeSparkAI(params.spark ?? {});
```
`?? {}` 应该不需要了

hezhijie0327 replied:
拿掉了

hezhijie0327 committed:
🔨 chore: cleanup code

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
🐛 fix: fix CI issue after merge

hezhijie0327 committed:
Merge branch 'lobehub:main' into spark

hezhijie0327 committed:
👷 build: add ENV

hezhijie0327 commented:
最近讯飞做了 Break Change... 改用了每个模型独立 `APIPassword`，原先 `APIKey:APISecret` 不再可用
![image](https://github.com/user-attachments/assets/7ae78416-9a67-45ea-b5fe-43fa2c3bcc3d)

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
♻️ refactor: support latest Spark HTTP SDK

hezhijie0327 commented:
Rollback 自定义 API 表单为通用 OpenAI 表单以适配最新的讯飞 HTTP SDK

吐槽一下，控制台中每个模型的 `APIPassword` 显示不同，但是实际通用，又是个神奇的 BUG
![image](https://github.com/user-attachments/assets/a08404f0-b471-4aba-b570-f92d55db2269)
![image](https://github.com/user-attachments/assets/5728d3ec-6ed6-44b4-89d8-f2656ab9d4ad)

hezhijie0327 committed:
♻️ refactor: cleanup

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
Merge branch 'main' into spark

hezhijie0327 committed:
🔨 chore: fix rebase conflicts

hezhijie0327 committed:
Merge branch 'main' into spark
